tests/nmea_parser.cc

```diff
@@ -63,3 +66,46 @@ TEST(NMEAParser, SampleGNZDASentence)
     EXPECT_EQ(time.tm_gmtoff, 0);
     EXPECT_EQ(time.tm_isdst, -1);
 }
+
+/*
+ * Check that a broken NMEA sentence triggers an NMEAError. Note that we append
+ * the correct checksum so that the parsing doesn't fail on those grounds.
+ */
+#define TEST_BROKEN_NMEA(name, line)                                                       \
```

Comment:
All the test magic also works inside functions so there's no need to use macros. Also, this functionality is built into gtest. You can do something like this:
```c++
class BrokenNMEATest : public ::testing::TestWithParam<std::string>
{
};

TEST_P(BrokenNMEATest, BrokenNMEA)
{
    NMEAParser nmea;
    GPSData data{};

    std::stringstream ss;
    const auto checksum = NMEAParser::computeChecksum(GetParam(), GetParam().length()); 
    ss << line << "*"
        << std::uppercase << std::setfill('0') << std::setw(2) << std::hex << checksum;

    EXPECT_THROW(nmea.parseCoordinates(ss.str(), data), NMEAError); 
}

INSTANTIATE_TEST_CASE_P(NMEAParser,
                        BrokenNMEA,
                        ::testing::Values(),);
```
Documented [here](https://github.com/google/googletest/blob/master/docs/advanced.md#value-parameterized-tests)

Reply:
That is cool, but here I do want to have different names for the different tests, so I know which of them is failing.

Reply:
Oh yeah - duh - I didn't grasp that aspect:
```c++
TEST(NMEAParser, TooFewFields){ nmeaDeathTest("$GNGGA,001043.00,4404.14036,N,12118.85961,W,1,12"); }
```
would probably still be nicer than a macro though

Reply:
True. I've done that now